etcdserver/server.go

```diff
@@ -928,6 +933,33 @@ func (s *EtcdServer) applyConfChange(cc raftpb.ConfChange, confState *raftpb.Con
 	return false, nil
 }
 
+func (s *EtcdServer) createRaftSnapshot(snapi uint64, confState raftpb.ConfState) {
+	clone := s.store.Clone()
+
+	// get the term of snapi before creating snapshot asynchronously, which
+	// ensures that the related entry is not compacted.
+	snapt, err := s.r.raftStorage.Term(snapi)
+	if err != nil {
+		log.Panicf("get term should never fail: %v", err)
+	}
+
+	go func() {
```

Comment:
is it necessary to run this in a go routine? i did not see it is necessary at least for now.